tests/functionaltests/Tests/CortanaTests.mm

```diff
@@ -106,9 +223,65 @@
         *kind = ActivationKind_VoiceCommand;
         return S_OK;
     });
-    
+
     fakeVoiceCommandActivatedEventArgs->Setget_PreviousExecutionState([](ApplicationExecutionState* state) {
         *state = ApplicationExecutionState_NotRunning;
         return S_OK;
     });
+
+    // Pass activation argument to method which activates the app
+    auto args = fakeVoiceCommandActivatedEventArgs.Detach();
+    UIApplicationActivationTest(reinterpret_cast<IInspectable*>(args), L"CortanaVoiceCommandForegroundTestDelegate");
+}
+
+TEST(CortanaTest, VoiceCommandForegroundActivationDelegateMethodsCalled) {
+    CortanaVoiceCommandForegroundTestDelegate* testDelegate = [[UIApplication sharedApplication] delegate];
+    NSDictionary* methodsCalled = [testDelegate methodsCalled];
+    ASSERT_TRUE(methodsCalled);
+    ASSERT_TRUE([methodsCalled objectForKey:@"application:willFinishLaunchingWithOptions:"]);
+    ASSERT_TRUE([methodsCalled objectForKey:@"application:didFinishLaunchingWithOptions:"]);
+    ASSERT_TRUE([methodsCalled objectForKey:@"application:didReceiveVoiceCommand:"]);
+}
+
+// Creates test method which we call in TEST_CLASS_SETUP to activate app
+TEST(CortanaTest, ProtocolForegroundActivation) {
+    LOG_INFO("CortanaTest Protocol Foreground Activation Test: ");
+
+    // Create mocked data to pass into application
+    auto fakeUri = Make<MockUri>();
+    fakeUri->SetToString([](HSTRING* text) {
+        Wrappers::HString value;
+        value.Set(L"CORTANA_TEST");
+        *text = value.Detach();
+        return S_OK;
+    });
+
+    auto fakeProtocolActivatedEventArgs = Make<MockProtocolActivatedEventArgs>();
+    fakeProtocolActivatedEventArgs->Setget_Uri([&fakeUri](IUriRuntimeClass** uri) {
+        fakeUri.CopyTo(uri);
+        return S_OK;
+    });
+
+    fakeProtocolActivatedEventArgs->Setget_Kind([](ActivationKind* kind) {
+        *kind = ActivationKind_Protocol;
+        return S_OK;
+    });
+
+    fakeProtocolActivatedEventArgs->Setget_PreviousExecutionState([](ApplicationExecutionState* state) {
+        *state = ApplicationExecutionState_NotRunning;
+        return S_OK;
+    });
+
+    // Pass activation argument to method which activates the app
+    auto args = fakeProtocolActivatedEventArgs.Detach();
+    UIApplicationActivationTest(reinterpret_cast<IInspectable*>(args), L"CortanaProtocolForegroundTestDelegate");
```

Comment:
yes, see my other comment in StarboardXaml.cpp for UIApplicationActivationTest function

---
In reply to: [69814747](https://github.com/Microsoft/WinObjC/pull/603#discussion_r69814747) [](ancestors = 69814747,69814325)